acceptance/continuous_load_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package acceptance
+
+import (
+	"flag"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/util/log"
+	"github.com/cockroachdb/cockroach/util/timeutil"
+)
+
+// Run tests in this file as follows:
+//
+// make test \
+//   PKG=./acceptance \
+//   TESTTIMEOUT=6h15m \
+//   TESTS=BlockWriter6Hours \
+//   TESTFLAGS='-v -remote -key-name google_compute_engine -cwd allocator_terraform'
+//
+// The SSH key you use for GCE must be in ~/.ssh/google_compute_engine. Also,
+// note the additional time in TESTTIMEOUT to allow time for the cluster to be
+// torn down.
+
+// continuousLoadTest generates continuous load against a remote test cluster
+// created specifically for the test.
+type continuousLoadTest struct {
+	// Prefix is prepended to all resources created by Terraform.
+	Prefix string
+	// NumNodes is the number of nodes in the test cluster.
+	NumNodes int
+	// Duration is the duration that the test should run for.
+	Duration time.Duration
+	// Process must be one of the processes (e.g. block_writer) that's
+	// downloaded by the Terraform configuration.
+	Process string
+	// CockroachDiskSizeGB is the size, in gigabytes, of the disks allocated
+	// for CockroachDB nodes. Leaving this as 0 accepts the default in the
+	// Terraform configs. This must be in GB, because Terraform only accepts
+	// disk size for GCE in GB.
+	CockroachDiskSizeGB int
+}
+
+// Run performs a continuous load test with the given parameters, checking the
+// health of the cluster regularly. Any failure of a CockroachDB node or load
+// generator constitutes a test failure.
+func (cl continuousLoadTest) Run(t *testing.T) {
+	if e := "GOOGLE_PROJECT"; os.Getenv(e) == "" {
+		t.Fatalf("%s environment variable must be set for Terraform", e)
+	}
+
+	// Ensure that the test timeout allows the load test to run and tear down
+	// properly, to avoid Terraform resource leaks.
+	const destroyTime = 10 * time.Minute
```

Comment:
Could you make this more general? We want all of the tests to (be able to) use this. Perhaps ```go
ctx := context.WithTestTimeout(parent context.Context) // has deadline set at (testduration-5min) (I think it's fair to just hard-code 5min for now).
```